test/unit/fiber_channel.cc

```diff
@@ -78,12 +78,103 @@ fiber_channel_get()
 	status = check_plan();
 }
 
+void
+fiber_channel_graceful_close_false(struct fiber_channel *channel)
+{
+	/* Test fiber_channel_graceful_close == false. */
+	char msg_1;
+	char msg_2;
+	char msg_3;
+	void *ptr = NULL;
+
+	ok(fiber_channel_put_timeout(channel, &msg_1, 0) == 0,
+	   "fiber_channel_put(msg_1)");
+
+	ok(fiber_channel_put_timeout(channel, &msg_2, 0) == 0,
+	   "fiber_channel_put(msg_2)");
+
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == &msg_1, "fiber_channel_get(1)");
+
+	fiber_channel_close(channel);
+
+	ok(channel->is_closed, "is_closed");
+	ok(channel->is_destroyed, "is_destroyed");
+
+	ok(fiber_channel_put_timeout(channel, &msg_3, 0) != 0,
+	   "not fiber_channel_put(msg_3)");
+
+	ptr = NULL;
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == NULL, "not fiber_channel_get(1)");
+
+	fiber_channel_delete(channel);
+}
+
+void
+fiber_channel_graceful_close_true(struct fiber_channel *channel)
+{
+	/* Test default fiber_channel_graceful_close == true. */
+	char msg_1;
+	char msg_2;
+	char msg_3;
+	void *ptr = NULL;
+
+	ok(fiber_channel_put_timeout(channel, &msg_1, 0) == 0,
+	   "fiber_channel_put(msg_1)");
+
+	ok(fiber_channel_put_timeout(channel, &msg_2, 0) == 0,
+	   "fiber_channel_put(msg_2)");
+
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == &msg_1, "fiber_channel_get(1)");
+
+	fiber_channel_close(channel);
+
+	ok(channel->is_closed, "is_closed");
+	ok(!channel->is_destroyed, "not is_destroyed");
+
+	ok(fiber_channel_put_timeout(channel, &msg_3, 0) != 0,
+	   "not fiber_channel_put(msg_3)");
+
+	ptr = NULL;
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == &msg_2, "fiber_channel_get(1)");
+	ok(channel->is_destroyed, "is_destroyed");
+
+	fiber_channel_delete(channel);
+}
+
+void
+fiber_channel_test_graceful_close()
+{
+	header();
+	plan(7 * 2 + 8);
+
+	struct fiber_channel *channel = fiber_channel_new(10);
+	fiber_channel_graceful_close_false(channel);
+	channel = fiber_channel_new(10);
+	fiber_channel_set_graceful_close(true);
+	fiber_channel_graceful_close_true(channel);
+	channel = fiber_channel_new(10);
+	fiber_channel_set_graceful_close(false);
+	fiber_channel_graceful_close_false(channel);
```

Comment:
Stilystic proposals:

* `fiber_channel_test_graceful_close()` -> `fiber_channel_test_close()` (because we test here not only graceful close).
* `fiber_channel_graceful_close_false()` -> `fiber_channel_forceful_close()` (or, better, just `test_forceful_close()`).
* `fiber_channel_graceful_close_true()` -> `fiber_channel_graceful_close()` (or, better, just `test_graceful_close()`).
* Mark functions as `static`.
* Create channels inside the `test_*ful_close()` functions.
* Split the cases by empty lines:

  ```c
  test_forceful_close();

  fiber_channel_set_graceful_close(true);
  test_graceful_close();

  fiber_channel_set_graceful_close(false);
  test_forceful_close();
  ```

Isn't it look easier?

Reply:
It does. Fixed.